### DISCUSSION TITLE
Assistance with setting up Sunbird RC Front-end and Back-end services

### DISCUSSION THREAD

**Discussion by mahesh-maney** [Sunbird-RC/community#186] [Issues]
Hi,
We have been trying to setup Sunbird RC front-end services as well as back-end services. We are following the documentation available on sunbird RC portal , however we have challenges setting up the services and infra.

We are trying out the Teacher / Student example as per the documentation and want to tryout the Front-end part https://docs.sunbirdrc.dev/use/setup-the-frontend#environment-config

We have the front-end running on docker but when we visit http://localhost/ it redirects me to http://localhost/install and just a blank page with errors on console - ERROR TypeError: Cannot read property 'keycloak' of null

Cloned the repository: git clone git@github.com:sunbird-rc/sunbird-rc-ui sunbird-rc/ui && cd $_
- Ran npm install
- Downloaded the schemas given [here](https://github.com/ref-registries/edu-core-registries/tree/main/ui-config/) and placed it in the src/assets/config/ folder. I also setup a registry using the corresponding schemas given [here](https://github.com/ref-registries/edu-core-registries/blob/main/schemas).
- Created a file - src/assets/config/config.json with the following content:

{
    "environment": "development",
    "baseUrl": "http://localhost:8081/api",
    "schemaUrl": "http://localhost:8081/api/docs/swagger.json",
    "keycloak": {
        "url": "https://localhost:8080/auth",
        "realm": "sunbird-rc",
        "clientId": "registry-frontend"
    }
}

- Ran npm run ng serve to start the application.
- Browsed to http://localhost:4200/ in the browser.
- This simply showed me a blank page, and no errors were logged in the console

Backend :
We are using the docker-compose file ( https://github.com/Sunbird-RC/sunbird-rc-core/blob/main/docker-compose.yml ) and following  ( https://docs.sunbirdrc.dev/using-the-apis ) to  tryout the  API's

Create/Authenticate/Retrieve/Update works fine but when we try make a claim we get
{
    "id": "sunbird-rc.registry.update",
    "ver": "1.0",
    "ets": 1643630462173,
    "params": {
        "resmsgid": "",
        "msgid": "f12da771-7094-4773-9158-ab1df9de2e11",
        "err": "",
        "status": "UNSUCCESSFUL"
    },
    "responseCode": "OK"
}

We need assistance in getting the claims API to work.

**Comment by coolbung** [comment]
@code-blooded-guy please help out with the frontend installation

**Reply by pramodkvarma** [reply]
I assume all of you have joined Discord channel as well.

**Reply by mahesh-maney** [reply]
yes, we have
